src/libNode/FinalBlockProcessing.cpp

```diff
@@ -533,6 +533,13 @@ void Node::PrepareGoodStateForFinalBlock() {
 bool Node::ProcessFinalBlock(const bytes& message, unsigned int offset,
                              [[gnu::unused]] const Peer& from) {
   LOG_MARKER();
+  return ProcessFinalBlockCore(message, offset, from);
+}
+
+bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
```

Comment:
is it necessary to split out a `Core` function for this case? I feel simply modify the `ProcessFinalBlock` shall be enough

Reply:
Yeah, Initially i had done same. But our instrucionHandler expects only 3 parameters. And i have to use 4th optionally.
https://github.com/Zilliqa/Zilliqa/blob/631a49e407081b8e9eb2c0ecc4d2777ec3e3940f/src/libNode/Node.cpp#L1973